### PR TITLE
Support popup auto resizing for dropDownBox with TreeView in Search mode (T724029)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/popup.js [hunk 1]
```diff
@@ -666,29 +672,54 @@ var Popup = Overlay.inherit({
...
-                    maxHeight: maxHeightValue
-                });
+            contentMinHeight = this._getOptionValue("minHeight", overlayContent);
+
+        var currentHeightStrategyClass = heightStrategies.static;
+
+        if(this._isAutoHeight() && this.option("autoResizeEnabled")) {
+            if(isAutoWidth || isOldSafari) {
+                if(isIE11) {
+                    currentHeightStrategyClass = heightStrategies.static;
```
Comment (DokaRus) [inline js/ui/popup.js]: You've already set this value to a variable at [R679](https://github.com/DevExpress/DevExtreme/pull/8340/files#diff-353a433c5241a480804acc2b08b8330fR679)

File: js/ui/popup.js [hunk 2]
```diff
@@ -666,29 +672,54 @@ var Popup = Overlay.inherit({
...
-                    minHeight: minHeightValue,
-                    maxHeight: maxHeightValue
-                });
+            contentMinHeight = this._getOptionValue("minHeight", overlayContent);
+
+        var currentHeightStrategyClass = heightStrategies.static;
+
+        if(this._isAutoHeight() && this.option("autoResizeEnabled")) {
+            if(isAutoWidth || isOldSafari) {
+                if(isIE11) {
```
Comment (DokaRus) [inline js/ui/popup.js]: Can you simplify it?

File: js/ui/popup.js [hunk 3]
```diff
@@ -666,29 +672,54 @@ var Popup = Overlay.inherit({
...
-                    maxHeightValue = sizeUtils.addOffsetToMaxHeight(contentMaxHeight, -toolbarsAndVerticalOffsetsHeight, container),
-                    minHeightValue = sizeUtils.addOffsetToMinHeight(contentMinHeight, -toolbarsAndVerticalOffsetsHeight, container);
-
-                cssStyles = extend(cssStyles, {
-                    minHeight: minHeightValue,
-                    maxHeight: maxHeightValue
-                });
+            contentMinHeight = this._getOptionValue("minHeight", overlayContent);
+
+        var currentHeightStrategyClass = heightStrategies.static;
```
Comment (DokaRus) [inline js/ui/popup.js]: Extract method

File: js/ui/popup.js [hunk 4]
```diff
@@ -46,6 +50,8 @@ var POPUP_CLASS = "dx-popup",
     BUTTON_CONTAINED_MODE = "contained";
 
 var isIE11 = (browser.msie && parseInt(browser.version) === 11);
+var isOldSafari = browser.safari && compareVersions(browser.version, [11]) < 0;
+var heightStrategies = { static: "", inherit: POPUP_CONTENT_INHERIT_HEIGHT_CLASS, flex: POPUP_CONTENT_FLEX_HEIGHT_CLASS };
```
Comment (DokaRus) [inline js/ui/popup.js]: Capitalize it like other constants

File: js/ui/popup.js [hunk 5]
```diff
@@ -666,29 +672,54 @@ var Popup = Overlay.inherit({
...
+                } else {
+                    currentHeightStrategyClass = heightStrategies.inherit;
+                }
+            } else {
+                currentHeightStrategyClass = heightStrategies.flex;
             }
-        } else {
+        }
+
+        if(currentHeightStrategyClass === heightStrategies.static) {
```
Comment (DokaRus) [inline js/ui/popup.js]: Extract method `getCssStyles`?

**Review by DokaRus — CHANGES_REQUESTED** [review]

**Review by mrjohn0011 — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

**Review by DokaRus — APPROVED** [review]

**Review by mrjohn0011 — APPROVED** [review]

**Pull request merged**